### DISCUSSION TITLE
Start - Stop - Restart KafkaBus

### DISCUSSION THREAD

**Discussion by schepersk** [Farfetch/kafkaflow#233] [Q&A]
I have a use case in which I have to be able to stop a running KafkaBus and restart it after some time, within the same application.
This is not possible right now, because all the providers are registered as Singleton and in the StartAsync method, the consumers are always being added (again).

A RestartAsync method would be very handy. Just start the existing configuration again.
I know Consumers can be paused. But this does not disconnect consumers group(s) from the Kafka broker.

Can someone help me with this please?

**Comment by filipeesch** [comment]
Hi @schepersk . 

We have the interface IConsumerAccessor that can be injected in any class to manage your consumers, when you get a consumer from there it returns an instance of MessageConsumer class, this class has a RestartAsync method, but would be very easy to implement a StartAsync() and StopAsync() method there (the RestartMethod just calls the internal StartAsync and SotpAsync). You can open a Pull Request with this improvement or we can implement it in the near future.
